src/core/mantle/evaluator/simple/utils.ts

```diff
@@ -0,0 +1,67 @@
+import { pick } from "lodash-es";
+
+import { Feature, StyleExpression } from "../../types";
+
+const JSONPATH_REGEX =
+  /\$[a-zA-Z_][a-zA-Z0-9_]*(\[[^\]]+\])?(\.[a-zA-Z_][a-zA-Z0-9_]*(\[[^\]]+\])?)*$/g;
+
+export function getCombinedReferences(expression: StyleExpression): string[] {
+  if (typeof expression === "string") {
+    return getReferences(expression);
+  } else {
+    const references: string[] = [];
+    for (const [condition, value] of expression.conditions) {
+      references.push(...getReferences(condition), ...getReferences(value));
+    }
+    return references;
+  }
+}
+
+export function getReferences(expression: string): string[] {
```

Comment:
I think this function is very heavy because this does same thing with replaceVariable.
So I think it's better to cache the result in here too like [I did](https://github.com/reearth/reearth-web/pull/574/files#diff-b985f02263b6110afa5cbeb5d637e627de98028bbcc80ed50c30ccbb2795980c).

Reply:
Done.